src/main/scala/mulAddSubRecodedFloatN.scala

```diff
@@ -296,6 +296,14 @@ class mulAddSubRecodedFloatN(sigWidth: Int, expWidth: Int, speed: Boolean = fals
           Mux(isNaNOut       , UInt(7 << expWidth-3), UInt(0, expWidth) )
   val fractOut = fractY | Fill(sigWidth, isNaNOut || isSatOut)
 
-  io.out := Cat(signOut, expOut, fractOut)
-  io.exceptionFlags := Cat(invalid, Bool(false), overflow, underflow, inexact)
+	val minPatchBypass1 = roundingMode_min && ~notSpecial_addZeros && signOut && (expOut(expWidth-1, expWidth-3) === UInt(0)) && totalUnderflowY && ~isZeroY	
+	val maxPatchBypass = roundingMode_max && ~notSpecial_addZeros && ~signOut && (expOut(expWidth-1, expWidth-3) === UInt(0)) && totalUnderflowY && ~isZeroY
+	val minPatchBypass2 = roundingMode_min && isZeroY && (signProd || opSignC) && ~notNaN_isInfOut 
+	val expOutPatched = Mux(maxPatchBypass || minPatchBypass1, UInt(minExp), expOut)
+	val signOutPatched = signOut || minPatchBypass2	
+	val outPatched = Cat(signOutPatched, expOutPatched, fractOut)
```

Comment:
Ah, I see, thanks.  I'll defer to John Hauser, arbiter of hardfloat.

On Mon, May 4, 2015 at 7:37 PM, ivanratkovic <notifications@github.com>
wrote:

> In src/main/scala/mulAddSubRecodedFloatN.scala
> <https://github.com/ucb-bar/berkeley-hardfloat/pull/3#discussion_r29640992>
> :
>
> > @@ -296,6 +296,14 @@ class mulAddSubRecodedFloatN(sigWidth: Int, expWidth: Int, speed: Boolean = fals
> >            Mux(isNaNOut       , UInt(7 << expWidth-3), UInt(0, expWidth) )
> >    val fractOut = fractY | Fill(sigWidth, isNaNOut || isSatOut)
> >
> > -  io.out := Cat(signOut, expOut, fractOut)
> > -  io.exceptionFlags := Cat(invalid, Bool(false), overflow, underflow, inexact)
> > +	val minPatchBypass1 = roundingMode_min && ~notSpecial_addZeros && signOut && (expOut(expWidth-1, expWidth-3) === UInt(0)) && totalUnderflowY && ~isZeroY	
> > +	val maxPatchBypass = roundingMode_max && ~notSpecial_addZeros && ~signOut && (expOut(expWidth-1, expWidth-3) === UInt(0)) && totalUnderflowY && ~isZeroY
> > +	val minPatchBypass2 = roundingMode_min && isZeroY && (signProd || opSignC) && ~notNaN_isInfOut
> > +	val expOutPatched = Mux(maxPatchBypass || minPatchBypass1, UInt(minExp), expOut)
> > +	val signOutPatched = signOut || minPatchBypass2	
> > +	val outPatched = Cat(signOutPatched, expOutPatched, fractOut)
>
> We test it using testfloat. It was not passing min and max, and after
> these fixes it passes all the tests.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ucb-bar/berkeley-hardfloat/pull/3/files#r29640992>.
>